### PR TITLE
Add ProductType to features and add IGS checks with access request verify limit

### PR DESCRIPTION
**Pull request by kimlisa**
part of https://github.com/gravitational/teleport.e/issues/2317

- Adds product type to help differentiate different usage based products
- adds igs check along with legacy check for `verifyAccessRequestMonthlyLimit`

### CODE REVIEW

File: lib/modules/modules.go [hunk 1]
```diff
@@ -174,6 +187,46 @@ func (f Features) ToProto() *proto.Features {
...
+	}
+}
+
+func GetAccessRequestFeatureConfig() AccessRequestsFeature {
+	return AccessRequestsFeature{
+		MonthlyRequestLimit: 5,
+	}
+}
+
+func GetDeviceTrustFeatureConfig() DeviceTrustFeature {
```
Comment (codingllama) [inline lib/modules/modules.go:223]: Could we add a godoc?

Could we rename this so it's clear that these are the default config for usage-based enterprise plans? Alternatively, should these live in e/ instead?

Same for similar methods.

**Review by codingllama — COMMENTED** [review]
Drive-by due to https://github.com/gravitational/teleport.e/pull/2763. Just looking at a couple of methods.

File: lib/modules/modules.go (hunk 1 above)
Comment (kimlisa) [reply lib/modules/modules.go:223]: moved it to enterprise

File: lib/modules/modules.go [hunk 2]
```diff
@@ -138,7 +148,10 @@ type AccessMonitoringFeature struct {
 
 // ToProto converts Features into proto.Features
 func (f Features) ToProto() *proto.Features {
+	productTypeValue := proto.ProductType_name[int32(f.ProductType)]
```
Comment (justinas) [inline lib/modules/modules.go:151]: This seems like an identity function (we convert e.g. `1` to `"PRODUCT_TYPE_TEAM"`, ten back to `1`). Can't we just do `proto.ProductType(f.ProductType)`, as long as the values in both "enums" match?

**Review by justinas — APPROVED** [review]

File: api/types/license.go [hunk 3]
```diff
@@ -123,6 +123,12 @@ type License interface {
 	GetAccountID() string
 
 	// GetFeatureSource returns where the features should be loaded from.
+	//
+	// Deprecated.
```
Comment (zmb3) [inline api/types/license.go:127]: Hmm, is this really deprecated? If we're still using it for the foreseeable future I dont think we should call it deprecated.

Who is going to update Cloud Enterprise to read features from Sales Center?

@bl-nero @capnspacehook are we good with making EUBP read from SC?

cc @bl-nero @capnspacehook @mcbattirola

File: lib/modules/modules.go [hunk 4]
```diff
@@ -174,6 +185,27 @@ func (f Features) ToProto() *proto.Features {
...
+const (
+	ProductTypeUnknown ProductType = 0
+	// ProductTypeTeam is Teleport ProductTypeTeam product.
+	ProductTypeTeam ProductType = 1
+	// ProductTypeEUB is Teleport Enterprise Usage Based product.
+	ProductTypeEUB ProductType = 2
+)
+
+// IsLegacy describes legacy product that existed before EUB product
+// was introduced.
```
Comment (zmb3) [inline lib/modules/modules.go:200]:
```suggestion
// IsLegacy describes the legacy enterprise product that existed before the
// usage-based product was introduced. Some features (Device Trust, for example)
// require the IGS add-on in usage-based products but are included for legacy
// licenses.
```

Some additional context can't hurt.

File: api/types/license.go (hunk 3 above)
Comment (kimlisa) [reply api/types/license.go:127]: > Who is going to update Cloud Enterprise to read features from Sales Center?

this [PR](https://github.com/gravitational/teleport.e/pull/2708) starts having `cloud enterprise` read from sales center, with this cloud enterprise will start periodically pinging salescenter for features

this [PR](https://github.com/gravitational/teleport.e/pull/2763) handles reading for EUB/Team

**Review by mcbattirola — APPROVED** [review]

**Review by zmb3 — APPROVED** [review]

**Pull request merged**